provider/agent.go

```diff
@@ -272,6 +260,23 @@ func agentResource() *schema.Resource {
 				Optional:    true,
 			},
 		},
+		CustomizeDiff: func(ctx context.Context, rd *schema.ResourceDiff, i interface{}) error {
+			if !rd.HasChange("metadata") {
+				return nil
+			}
+
+			keys := map[string]struct{}{}
```

Comment:
Nit: If you change this to a `map[string]bool` you can make the below condition look a lot nicer, at the expense of a few bytes:

```
if keys[key] {
  return ...
}
```